### PR TITLE
[API-41169][API-41170] Move POA dependent assignment into async job

### PR DESCRIPTION
**Pull request by tycol7**
## Summary

- Move v1/v2 dependent POA assignments into async (Sidekiq) job
- Add logic to either assign the dependent POA or assign the veteran POA (but not both)

## Related issue(s)

- [API-41169](https://jira.devops.va.gov/browse/API-41169)
- [API-41170](https://jira.devops.va.gov/browse/API-41170)

## Testing done

- [x] *New code is covered by unit tests*

## Screenshots

N/A

## What areas of the site does it impact?

- v1/v2 2122(a) controllers
- New Sidekiq job 

## Acceptance criteria

- [x]  I updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

See below.

### CODE REVIEW

**Comment by tycol7** [comment]
For v1, we save the `PowerOfAttorney` regardless of whether or not there is a dependent claimant:

https://github.com/department-of-veterans-affairs/vets-api/blob/8bbe3f0ec7c63d0bdc3874bf9bcb2ddc3d3a9180/modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb#L37-L53

Do we want to add logic to skip this step if there is a dependent claimant?

File: modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb [hunk 1]
```diff
@@ -55,7 +54,12 @@ def submit_form_2122 # rubocop:disable Metrics/MethodLength
 
           data = power_of_attorney.form_data
 
-          if data.dig('signatures', 'veteran').present? && data.dig('signatures', 'representative').present?
+          if feature_enabled_and_claimant_present?
+            ClaimsApi::PoaAssignDependentClaimantJob.perform_async(
+              dependent_claimant_poa_assignment_service(poa_code:, file_number:, dependent_participant_id:,
+                                                        claimant_ssn:)
+            )
```
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb:61]: how does this render in the sidekiq UI? wondering if we run the risk of showing the file_number, PID, SSN, etc;

Comment (tycol7) [reply modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb:61]: No PII, just the `DependentClaimantPoaAssignmentService` object ID:

sidekiq

**Review by FonzMP — DISMISSED** [review]
:shipit:

**Comment by FonzMP** [comment]
actually @tycol7 , does this build the PoA Document for the Claimant? I don't know if we've covered that scenario? Won't we need the physical 21-22 in the respective eFolder (Veteran's)? I'm not sure that's called out in the tickets but I feel like that's a miss currently?

**Comment by tycol7** [comment]
> actually @tycol7 , does this build the PoA Document for the Claimant? I don't know if we've covered that scenario? Won't we need the physical 21-22 in the respective eFolder (Veteran's)? I'm not sure that's called out in the tickets but I feel like that's a miss currently?

I don't think we currently build the PoA document for the claimant, but we can always add it in another ticket 🤷

**Review by FonzMP — APPROVED** [review]
:shipit:

**Pull request merged**